src/ponyjs/models/base.js

```diff
@@ -56,14 +53,12 @@ class ModelState {
 
 
 class ModelBase {
-  constructor(data={}) {
+  constructor(data) {
 
     // set up the instance state
     this._state = data;
 
-    for (let key of Object.keys(data)) {
-      this[key] = data[key]
-    }
+    Object.assign(this, data);
```

Comment:
I wonder what happens now if `data` is `undefined`, do you know if there's a test for that case?